markdownlint/TOP005_blanksAroundMultilineHtmlTags/TOP005_blanksAroundMultilineHtmlTags.js

```diff
@@ -68,10 +68,8 @@ module.exports = {
        * +1 required as file line numbers are 1-indexed.
        */
       onError({
-        lineNumber: lineNumber + 1,
-        detail: `\n  Expected blank lines/code block delimiters: { Before: 1, After: 1 }\n  Actual blank lines/code block delimiters: { Before: ${
-          lineBeforeIsValid ? 1 : 0
-        }, After: ${lineAfterIsValid ? 1 : 0} }\n`,
+        lineNumber: lineNumber + 1,      
+        detail: `Expected a blank line or a code block delimiter (\`\`\`)${lineBeforeIsValid ? "" : " before the tag"}${lineBeforeIsValid || lineAfterIsValid ? "" : " and"}${lineAfterIsValid ? "" : " after the tag"}`,
```

Comment:
This isn't quite what I meant. What you had before with the if conditionals was fine, it was just that the string formation was handled via concatenation and not template literals, which is inconsistent with how the same logic right before it handled things.

What I meant was:
```js
      if (!lineBeforeIsValid) {
        errorMessage = `${errorMessage} before the tag`;
      }
      if (!lineBeforeIsValid && !lineAfterIsValid) {
        errorMessage = `${errorMessage} and`;
      }
      if (!lineAfterIsValid) {
        errorMessage = `${errorMessage} after the tag`;
      }
```

Which follows the exact same approach as the replacementText bit before. Else you'd have the same logic using template literals to form a string then suddenly concatenation, when they should be the same.

Reply:
> This isn't quite what I meant. What you had before with the if conditionals was fine, it was just that the string formation was handled via concatenation and not template literals, which is inconsistent with how the same logic right before it handled things.
> 
> What I meant was:
> 
> ```js
>       if (!lineBeforeIsValid) {
>         errorMessage = `${errorMessage} before the tag`;
>       }
>       if (!lineBeforeIsValid && !lineAfterIsValid) {
>         errorMessage = `${errorMessage} and`;
>       }
>       if (!lineAfterIsValid) {
>         errorMessage = `${errorMessage} after the tag`;
>       }
> ```
> 
> Which follows the exact same approach as the replacementText bit before. Else you'd have the same logic using template literals to form a string then suddenly concatenation, when they should be the same.

Ah :laughing: , this makes more sense now. Thank you for clarifying!